.github/workflows/build.yml

```diff
@@ -0,0 +1,45 @@
+name: Build
+on:
+  push:
+    branches: [master, branch-*, dogfood-*]
+  pull_request:
+  merge_group:
```

Comment:
Out of curiosity, on the [sonar-custom-plugin-example repo](https://github.com/SonarSource/sonar-custom-plugin-example/blob/10.x/.github/workflows/build.yml), we have different triggers:

```
  pull_request:
    types: [opened, synchronize, reopened]
  merge_group:
    types: [checks_requested]
```

Why is it different?